### PR TITLE
[ACS-5308] Highlight the selected filter in ADW

### PR DESCRIPTION
**Pull request by Yasa-Nataliya**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**
When I click on any menu item in files this adf-active class does not remove, so I have added an active class based on the route.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
https://alfresco.atlassian.net/browse/ACS-5306.
There is a dependency between this PR and this one: https://github.com/Alfresco/alfresco-applications/pull/241.

### CODE REVIEW

File: lib/process-services/src/lib/process-list/components/process-filters.component.ts [hunk 1]
```diff
@@ -96,6 +109,10 @@ export class ProcessFiltersComponent implements OnInit, OnChanges {
         }
     }
 
+    isProcessRoute(filter: ProcessInstanceFilterRepresentation) : boolean {
+        return this.currentRoute.includes('process') && this.currentFilter === filter;
```
Comment (DenysVuika) [inline lib/process-services/src/lib/process-list/components/process-filters.component.ts:113]: you can do this in the "subcribe" section on the init, and assign to the Boolean property, without calculating each time from the html template

File: lib/process-services/src/lib/process-list/components/process-filters.component.ts [hunk 2]
```diff
@@ -71,15 +73,26 @@ export class ProcessFiltersComponent implements OnInit, OnChanges {
...
 
     constructor(private processFilterService: ProcessFilterService,
-                private appsProcessService: AppsProcessService) {
+                private appsProcessService: AppsProcessService,
+                private router: Router) {
     }
 
     ngOnInit() {
-        this.iconsMDL = new IconModel();
+        this.iconsMDL = new IconModel();this.router.events
```
Comment (DenysVuika) [inline lib/process-services/src/lib/process-list/components/process-filters.component.ts:87]: missing new line

File: lib/process-services/src/lib/task-list/components/task-filters.component.ts [hunk 3]
```diff
@@ -95,6 +109,10 @@ export class TaskFiltersComponent implements OnInit, OnChanges {
         }
     }
 
+    isTaskRoute(filter: FilterParamsModel) : boolean {
+        return this.currentRoute.includes('task') && this.currentFilter === filter;
```
Comment (DenysVuika) [inline lib/process-services/src/lib/task-list/components/task-filters.component.ts:113]: same as for the process filter, evaluate the flag inside `.subscribe((navigationStart: NavigationStart)`

File: lib/process-services/src/lib/process-list/components/process-filters.component.ts [hunk 4]
```diff
@@ -71,15 +73,29 @@ export class ProcessFiltersComponent implements OnInit, OnChanges {
...
 
     ngOnInit() {
         this.iconsMDL = new IconModel();
+        this.router.events
+        .pipe(
+            filter((event) => event instanceof NavigationStart),
+            takeUntil(this.onDestroy$)
+        )
+        .subscribe((navigationStart: NavigationStart) => {
+            this.currentRoute = navigationStart.url;
```
Comment (DenysVuika) [inline lib/process-services/src/lib/process-list/components/process-filters.component.ts:95]: you probably don't need this variable, as component is not using "currentRoute" other than on the next line

File: lib/process-services/src/lib/task-list/components/task-filters.component.ts [hunk 5]
```diff
@@ -71,15 +73,29 @@ export class TaskFiltersComponent implements OnInit, OnChanges {
...
 
     ngOnInit() {
         this.iconsMDL = new IconModel();
+        this.router.events
+            .pipe(
+                filter((event) => event instanceof NavigationStart),
+                takeUntil(this.onDestroy$)
+            )
+            .subscribe((navigationStart: NavigationStart) => {
+                this.currentRoute = navigationStart.url;
```
Comment (DenysVuika) [inline lib/process-services/src/lib/task-list/components/task-filters.component.ts:96]: you can replace with local variable
```suggestion
                const currentRoute = navigationStart.url;
```

File: lib/process-services/src/lib/process-list/components/process-filters.component.ts [hunk 6]
```diff
@@ -71,15 +73,29 @@ export class ProcessFiltersComponent implements OnInit, OnChanges {
 
     filters: UserProcessInstanceFilterRepresentation [] = [];
     active = false;
+    currentRoute: string= '';
```
Comment (DenysVuika) [inline lib/process-services/src/lib/process-list/components/process-filters.component.ts:76]: not used anywhere

**Review by DenysVuika — CHANGES_REQUESTED** [review]

**Comment by DenysVuika** [comment]
@Yasa-Nataliya please add unit tests for the changed/added code

File: lib/process-services/src/lib/task-list/components/task-filters.component.ts (hunk 5 above)
Comment (Yasa-Nataliya) [reply lib/process-services/src/lib/task-list/components/task-filters.component.ts]: Replaced

**Review by DenysVuika — APPROVED** [review]

File: lib/process-services/src/lib/task-list/components/task-filters.component.ts [hunk 7]
```diff
@@ -71,15 +73,29 @@ export class TaskFiltersComponent implements OnInit, OnChanges {
 
     filters: FilterRepresentationModel [] = [];
 
+    private onDestroy$ = new Subject<boolean>();
+    currentRoute: string = '';
```
[the remark this answers is not in the document]
Comment (Yasa-Nataliya) [reply lib/process-services/src/lib/task-list/components/task-filters.component.ts]: I have removed it

**Comment by DenysVuika** [comment]
@Yasa-Nataliya I have fixed some code issues in your PR, but you need to address the failing e2e caused by this PR

**Comment by Yasa-Nataliya** [comment]
> @Yasa-Nataliya I have fixed some code issues in your PR, but you need to address the failing e2e caused by this PR

Thanks @DenysVuika, I'll take a look and address the failing e2e test

**Review by DenysVuika — APPROVED** [review]

**Pull request merged**